File: app/src/main/java/com/pietrantuono/podcasts/main/presenter/MainPresenter.java
package com.pietrantuono.podcasts.main.presenter;

import com.pietrantuono.podcasts.GenericPresenter;
import com.pietrantuono.podcasts.main.view.MainView;

public class MainPresenter implements GenericPresenter {
    private MainView view;

    public void bindView(MainView view) {
        this.view = view;
    }


    @Override
    public void onDestroy() {
    }

    public void onAddPodcastSelected() {
        view.navigateToAddPodcast();
    }

    @Override
    public void onPause() {
    }

    @Override
    public void onResume() {

    }

    public void onCreate(boolean activityRecreated) {
        if (!activityRecreated) {
            view.navigateToSubscribedPodcasts();
        }
    }

    public void onSubscribeSelected() {
        view.navigateToSubscribedPodcasts();
    }
}

File: app/src/main/java/com/pietrantuono/podcasts/main/presenter/MainPresenter.kt
package com.pietrantuono.podcasts.main.presenter

import com.pietrantuono.podcasts.GenericPresenter
import com.pietrantuono.podcasts.main.view.MainView

class MainPresenter(private val killSwich: KillSwitch) : GenericPresenter {

    private var view: MainView? = null

    fun bindView(view: MainView) {
        this.view = view
    }

    override fun onDestroy() {}

    fun onAddPodcastSelected() {
        view?.navigateToAddPodcast()
    }

    override fun onPause() {
        killSwich.unsubscribe()
    }

    override fun onResume() {
        killSwich.checkIfNeedsToBeKilled(view)
    }

    fun onCreate(activityRecreated: Boolean) {
        if (!activityRecreated) {
            view?.navigateToSubscribedPodcasts()
        }
    }

    fun onSubscribeSelected() {
        view?.navigateToSubscribedPodcasts()
    }

    fun onSettingsSelected() {
        view?.navigateToSettings()
    }

    fun onDownloadsSelected() {
        view?.navigateToDownloads()
    }

}

